Enhancer/ContentRepositoryEnhancer.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/*
+ * This file is part of the Symfony CMF package.
+ *
+ * (c) 2011-2015 Symfony CMF
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Cmf\Component\Routing\Enhancer;
+
+use Symfony\Cmf\Component\Routing\ContentRepositoryInterface;
+use Symfony\Cmf\Component\Routing\RouteObjectInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * This enhancer sets the content to target field if the route has content id.
+ *
+ * Works with ContentRepositoryInterface that you can search the content.
+ *
+ * @author Samusev Andrey
+ */
+class ContentRepositoryEnhancer implements RouteEnhancerInterface
+{
+    /**
+     * @var ContentRepositoryInterface
+     */
+    private $contentRepository;
+
+    /**
+     * @var string
+     */
+    private $target;
+
+    /**
+     * @var string
+     */
+    private $source;
+
+    /**
+     * @param ContentRepositoryInterface $contentRepository repository for the search content
+     * @param string                     $target            the field name of the content id
+     * @param string                     $source            the field name to set from the map
```

Comment:
seems like these 2 descriptions should be swapped (`$target` is the new default value and `$source` is the name of the content id).